sdk/clientcore/core/src/main/java/io/clientcore/core/http/RestProxy.java

```diff
@@ -68,34 +68,44 @@ public Object invoke(Object proxy, final Method method, Object[] args) {
      * @param swaggerInterface the Swagger interface to provide a proxy implementation for
      * @param httpPipeline the HttpPipelinePolicy and HttpClient pipeline that will be used to send Http requests
      * @param <A> the type of the Swagger interface
+     * @param useRestProxy flag to enable RestProxy functionality
      * @return a proxy implementation of the provided Swagger interface
      */
     @SuppressWarnings("unchecked")
-    public static <A> A create(Class<A> swaggerInterface, HttpPipeline httpPipeline) {
-        final SwaggerInterfaceParser interfaceParser = SwaggerInterfaceParser.getInstance(swaggerInterface);
-        final RestProxy restProxy
-            = new RestProxy(httpPipeline, interfaceParser, new JsonSerializer(), new XmlSerializer());
+    public static <A> A create(Class<A> swaggerInterface, HttpPipeline httpPipeline, boolean useRestProxy) {
```

Comment:
@srnagar I added it to the public API so that we can have the ServiceClientImpl generation to on/off the flag but that make it a part of generation utility too, Thoughts?